src/Data/Subst.hs

```diff
@@ -26,9 +26,9 @@ substDelete name = Subst . filter ((/= name) . fst) . getSubst
 substSingleton :: Name -> value -> Subst value
 substSingleton name value = Subst [(name, value)]
 
-substExtend :: Binder value => Name -> value -> Subst value -> Subst value
-substExtend name value = (substSingleton name value <>)
+substExtend :: Binder value value => Name -> value -> Subst value -> Subst value
+substExtend name value = (<> substSingleton name value)
 
 
-class Binder value where
-  substitute :: Subst value -> value -> value
+class Binder ty value where
+  substitute :: Subst ty -> value -> value
```

Comment:
Extending `Binder` thus allows us to substitute inside of `Error` and `ElabTerm`.